ast/parser_ext.go

```diff
@@ -496,22 +495,28 @@ func ParseStatements(filename, input string) ([]Statement, []*Comment, error) {
 	return stmts, comments, postProcess(filename, stmts)
 }
 
-func convertErrList(filename string, errs errList) error {
+func formatParserErrors(filename string, bs []byte, err error) error {
+	// Errors returned by the parser are always of type errList and the errList
+	// always contains *parserError.
+	// https://godoc.org/github.com/mna/pigeon#hdr-Error_reporting.
+	errs := err.(errList)
 	r := make(Errors, len(errs))
 	for i, e := range errs {
-		switch e := e.(type) {
-		case *parserError:
-			r[i] = formatParserError(filename, e)
-		default:
-			r[i] = NewError(ParseErr, nil, e.Error())
-		}
+		r[i] = formatParserError(filename, bs, e.(*parserError))
 	}
 	return r
 }
 
-func formatParserError(filename string, e *parserError) *Error {
+func formatParserError(filename string, bs []byte, e *parserError) *Error {
 	loc := NewLocation(nil, filename, e.pos.line, e.pos.col)
-	return NewError(ParseErr, loc, e.Inner.Error())
+	inner := e.Inner.Error()
+	idx := strings.Index(inner, "no match found")
+	if idx >= 0 {
+		inner = inner[:idx+14]
```

Comment:
What is the significance of `14` ?

Reply:
len("no match found") = 14. I'll add a comment.